adagrams/game.py

```diff
@@ -1,11 +1,144 @@
+import random
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+
 def draw_letters():
-    pass
+    tiles = [] #list that's ten letters
+    count = 0 #going to stop at 10
+    letters_dict = LETTER_POOL.copy()
+    while count != 10:
+        letter = random.choice(list(letters_dict.keys()))
+        if letters_dict[letter] > 0:
+            letters_dict[letter] -= 1
+            tiles.append(letter)
+            count += 1
+        else:
+            continue
+    return tiles
+        
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letter_bank_dict = {}
+    word_dict = {}
+    word = word.upper()
+
+    for letter in letter_bank:
+        if letter not in letter_bank_dict:
+            letter_bank_dict[letter] = 1
+        else:
+            letter_bank_dict[letter] += 1
 
+    for character in word:
+        if character not in letter_bank:
+            return False
+        else:
+            if character not in word_dict:
+                word_dict[character] = 1
+            else:
+                word_dict[character] += 1
+
+    for characters in word_dict:
+        if word_dict[characters] > letter_bank_dict[characters]:
+            return False
```

Comment:
I see that you create two dictionaries `letter_bank_dict` which is a frequency of how many times the letters in the list `letter_bank` appear and you also have `word_dict` which tells us how many times a letter in `word` appears. 

Do we need to create two separate dictionaries in order to see if `word` uses the available letters in `letter_bank`? I think we can get away without needing to create several data structures.

While this isn't a huge issue, it does add complexity because other coders need to figure out why we have several data structures and how they work together. Also, we haven't covered this yet but in our Big O lesson, we'll see how creating data structures whose size grows in relation to a function's input can effect performance. 

What if we took a different approach? You could still leverage letter_bank_dict and use this dictionary to check that the letters in word are available. 

```python
    for letter in letter_bank:
        if letter not in letter_bank_dict:
            letter_bank_dict[letter] = 1
        else:
            letter_bank_dict[letter] += 1

    for letter in word:
        if letter not in letter_bank:
            return False
        else:
            if letter_bank_dict[letter] > 0:
                letter_bank_dict[letter] = letter_bank_dict[letter] - 1
            else: 
                return False
    
    return True
```